src/structures/message.ts

```diff
@@ -115,6 +116,7 @@ export class Message extends SnowflakeBase {
     this.application = data.application ?? this.application
     this.messageReference = data.message_reference ?? this.messageReference
     this.flags = data.flags ?? this.flags
+    this.url = `https://discord.com/channels/${data.guild_id || "@me"}/${data.channel_id}/${this.id}`
```

Comment:
i think you should use `this.guildID` and `this.channelID` for this because data could also have no channel_id

Reply:
fixed